hdfs/client.py

```diff
@@ -986,10 +987,7 @@ class InsecureClient(Client):
 
   def __init__(self, url, user=None, **kwargs):
     user = user or getuser()
-    session = kwargs.setdefault('session', rq.Session())
-    if not session.params:
-      session.params = {}
-    session.params['user.name'] = user
+    kwargs.setdefault('params', {})['user.name'] = user
```

Comment:
For subclasses of clients, let's keep the previous behavior. After this PR is merged, we should be in a state where:

+ The base `Client` class doesn't edit its `session` (satisfying your use-case).
+ Subclasses add convenience functionality which sets up the session appropriately (therefore modifying/owning it).